test/demos/reversediffzero.jl

```diff
@@ -0,0 +1,132 @@
+"The simplest viable reverse mode a AD, only supports `Float64`"
+module ReverseDiffZero
+using ChainRulesCore
+using Test
+
+struct Tracked <: Real
+    propagate::Function
+    primal::Float64
+    tape::Vector{Any}  # a reference to a shared tape
+    grad::Base.RefValue{Float64} # current accumulated gradient
```

Comment:
Maybe don't call this `grad`. Perhaps `cotangent` or `sensitivity`?

Reply:
true, I think Tracker.jl calls it `grad`. but it isn't really that.